lib/generator.js

```diff
@@ -155,13 +155,19 @@ class Generator {
    *   console.error(e);
    * }
    *
-   * @param  {AsyncAPIDocument} asyncapiDocument AsyncAPIDocument object to use as source.
+   * @param  {AsyncAPIDocument | string} asyncapiDocument AsyncAPIDocument object to use as source.
+   * @param  {Object} [parseOptions={}] AsyncAPI Parser parse options. Check out {@link https://www.github.com/asyncapi/parser-js|@asyncapi/parser} for more information. Remember to use the right options to the right parser depending on the template you are using.
    * @return {Promise}
    */
-  async generate(asyncapiDocument) {
-    if (!isAsyncAPIDocument(asyncapiDocument)) throw new Error('Parameter "asyncapiDocument" must be an AsyncAPIDocument object.');
-
+  // NOSONAR
+  async generate(asyncapiDocument, parseOptions = {}) {
+    const isAlreadyParsedDocument = isAsyncAPIDocument(asyncapiDocument);
+    const isParsableCompatible = asyncapiDocument || typeof asyncapiString === 'string';
```

Comment:
where is `asyncapiString` coming from 🤔

Reply:
Nice catch, was actually a bug 😆 

Added a test to catch it.